barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.hpp

```diff
@@ -45,8 +53,12 @@ class ClientIVC {
 
     Proof prove();
 
-    bool verify(Proof& proof);
+    bool verify(Proof& proof, const std::vector<VerifierAccumulator>& verifier_instances);
 
     HonkProof decider_prove() const;
+
+    void decider_prove_and_verify(const VerifierAccumulator&) const;
+
+    void precompute_folding_verification_keys();
```

Comment:
Soo maybe it's odd that precomputation is here but ClientIVC depends on `GoblinMockCircuits` (we need some initial ecc ops because of the zero commitments problem) so if we moved this function  there (which might make more sense since we precompute the vk of circuits from `GoblinMockCircuits`) we create a circular dependency.